### PR TITLE
[SECT-1483] Adding scanner_type for each scanner

### PR DESCRIPTION
**Pull request by swkanwar**
Adding support for scanner_type for each scanner

### CODE REVIEW

File: lib/salus/scanners/base.rb [hunk 1]
```diff
@@ -53,6 +53,10 @@ def self.supported_languages
       []
     end
 
+    def self.scanner_type
```
Comment (saikat056) [inline lib/salus/scanners/base.rb:56]:
```ruby
def self.scanner_type
  raise NoMethodError, 'implement in subclass'
end
```

Usually, in ruby, we force the developers to implement the method in subcalss by raising an exception. We have similar code in other [places](https://github.com/coinbase/salus/blob/9c93eb2e2b9d87b4a60de7bc1512c4f0556772ce/lib/salus/scanners/node_audit.rb#L143).

**Comment by saikat056** [comment]
We can create a class of scanner_type constants and refer the constant in each scanner.

```ruby
module Salus::ScannerTypes
  class Types
       SAST = 'SAST'.freeze
       DAST = 'DAST'.freeze
  end
end
```

Then, refer to the constants from the scanner sub-class

```ruby
def self.scanner_type
   Salus::ScannerTypes::Types::SAST
end
```

File: lib/salus/scanners/brakeman.rb [hunk 2]
```diff
@@ -6,6 +6,10 @@
 
 module Salus::Scanners
   class Brakeman < Base
+    def self.scanner_type
+      'SAST'
+    end
```
Comment (saikat056) [inline lib/salus/scanners/brakeman.rb:11]: If we have a class defining all [constants](https://github.com/coinbase/salus/pull/631#issuecomment-1151534600), then we can write this part as follows:

```ruby
def self.scanner_type
  Salus::ScannerTypes::Types::SAST
end
```

File: lib/salus/scanners/base.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (swkanwar) [reply lib/salus/scanners/base.rb:56]: Done, thank you!

Comment (swkanwar) [reply lib/salus/scanners/base.rb:56]: Done

**Review by saikat056 — APPROVED** [review]

**Pull request merged**